### PR TITLE
Add `halt` command

### PR DESCRIPTION
**Pull request by MWDelaney**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -58,6 +59,9 @@ func main() {
 		"up": func() (cli.Command, error) {
 			return cmd.NewUpCommand(ui, trellis), nil
 		},
+		"halt": func() (cli.Command, error) {
```
Comment (austinpray) [inline main.go]: bet you won't alias "down" to "halt"

Comment (MWDelaney) [reply main.go]: Bet I would.

File: cmd/halt.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	"trellis-cli/trellis"
+
+	"github.com/mitchellh/cli"
+)
+
+type HaltCommand struct {
+	UI          cli.Ui
+	Trellis     *trellis.Trellis
+	flags       *flag.FlagSet
+	noGalaxy    bool
```
Comment (swalkinshaw) [inline cmd/halt.go]: These aren't needed

File: cmd/halt.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	noProvision bool
+}
+
+func NewHaltCommand(ui cli.Ui, trellis *trellis.Trellis) *HaltCommand {
+	c := &HaltCommand{UI: ui, Trellis: trellis}
+	c.init()
+	return c
+}
+
+func (c *HaltCommand) init() {
```
Comment (swalkinshaw) [inline cmd/halt.go]: Since there's no flags, you can remove this function entirely

File: cmd/halt.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+type HaltCommand struct {
+	UI          cli.Ui
+	Trellis     *trellis.Trellis
+	flags       *flag.FlagSet
+	noGalaxy    bool
+	noProvision bool
+}
+
+func NewHaltCommand(ui cli.Ui, trellis *trellis.Trellis) *HaltCommand {
```
Comment (swalkinshaw) [inline cmd/halt.go]: Since there's no `init` function needed, you can remove this function entirely too

File: cmd/halt.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	c.flags.Usage = func() { c.UI.Info(c.Help()) }
+}
+
+func (c *HaltCommand) Run(args []string) int {
+	if err := c.Trellis.LoadProject(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	if err := c.flags.Parse(args); err != nil {
```
Comment (swalkinshaw) [inline cmd/halt.go]: ✂️

File: cmd/halt.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+	err := vagrantHalt.Run()
+	if err != nil {
+		return 1
+	}
+
+	return 0
+}
+
+func (c *HaltCommand) Synopsis() string {
+	return "Starts and provisions the Vagrant environment by running 'vagrant halt'"
```
Comment (swalkinshaw) [inline cmd/halt.go]: This needs updating

File: main.go [hunk 7]
```diff
@@ -58,6 +59,9 @@ func main() {
 		"up": func() (cli.Command, error) {
 			return cmd.NewUpCommand(ui, trellis), nil
 		},
+		"down": func() (cli.Command, error) {
+			return cmd.NewHaltCommand(ui, trellis), nil
```
Comment (swalkinshaw) [inline main.go]:
```go
return &cmd.HaltCommand{UI: ui, Trellis: trellis}, nil
```

**Comment by swalkinshaw** [comment]
See https://github.com/roots/trellis-cli/pull/43

**Pull request closed without merging**